paddle/fluid/operators/softmax_op.h

```diff
@@ -30,10 +31,15 @@ class SoftmaxKernel : public framework::OpKernel<T> {
 
     // allocate memory on device.
     Out->mutable_data<T>(context.GetPlace());
-
     int rank = X->dims().size();
     Tensor X_2d = framework::ReshapeToMatrix(*X, rank - 1);
     Tensor Out_2d = framework::ReshapeToMatrix(*Out, rank - 1);
+    if (std::type_index(typeid(T)) ==
+        std::type_index(typeid(platform::float16))) {
```

Comment:
Why can we just use `std::is_same<T, platform::float16>` here? It may be faster.

Reply:
good point. fixed/